tests/frontend/components/test_build_input_section.py

```diff
@@ -5,161 +5,205 @@
 import ssb_dash_components as ssb  # type: ignore[import-untyped]
 from datadoc_model import model
 
-from datadoc import enums
 from datadoc.enums import SupportedLanguages
 from datadoc.frontend.components.builders import build_input_field_section
-from datadoc.frontend.fields.display_base import get_enum_options_for_language
+from datadoc.frontend.fields.display_base import VariablesCheckboxField
+from datadoc.frontend.fields.display_base import VariablesDropdownField
+from datadoc.frontend.fields.display_base import VariablesInputField
+from datadoc.frontend.fields.display_base import VariablesPeriodField
 from datadoc.frontend.fields.display_variables import OBLIGATORY_VARIABLES_METADATA
 from datadoc.frontend.fields.display_variables import OPTIONAL_VARIABLES_METADATA
 
+VARIABLES_METADATA = OBLIGATORY_VARIABLES_METADATA + OPTIONAL_VARIABLES_METADATA
+TEST_VARIABLE = model.Variable(short_name="sykepenger")
+
 INPUT_FIELD_SECTION = [
     build_input_field_section(
-        OBLIGATORY_VARIABLES_METADATA,
+        VARIABLES_METADATA,
         model.Variable(short_name="hoveddiagnose"),
         SupportedLanguages.NORSK_NYNORSK,
     ),
     build_input_field_section(
-        OBLIGATORY_VARIABLES_METADATA,
+        VARIABLES_METADATA,
         model.Variable(short_name="pers_id"),
         SupportedLanguages.NORSK_BOKMÅL,
     ),
     build_input_field_section(
-        OBLIGATORY_VARIABLES_METADATA,
+        VARIABLES_METADATA,
         model.Variable(short_name="ber_bruttoformue"),
         SupportedLanguages.ENGLISH,
     ),
     build_input_field_section(
-        OBLIGATORY_VARIABLES_METADATA,
+        VARIABLES_METADATA,
         model.Variable(short_name="sykepenger"),
         SupportedLanguages.NORSK_BOKMÅL,
     ),
 ]
 
 
-def test_build_input_field_section_no_input_return_empty_list():
-    """Test build with empty inputs."""
+def test_build_input_field_section_no_input():
+    """Assert build with empty inputs returns dash bootstrap Form component and empty children list."""
     input_section = build_input_field_section(
         [],
         "",
         "",
     )
     assert input_section.children == []
+    assert isinstance(input_section, dbc.Form)
+
+
+@pytest.mark.parametrize(
+    ("input_field_section"),
+    INPUT_FIELD_SECTION,
+)
+def test_build_input_fields_input_components(input_field_section):
+    """Assert fields input field for ."""
+    type_input = ssb.Input
+    elements_of_input = list(
+        filter(lambda x: isinstance(x, type_input), input_field_section.children),
```

Comment:
it's "more pythonic" to use a list comprehension in these situations, but they're functionally identical.